client/src/main/java/io/pravega/client/tables/impl/MapWrapperImpl.java

```diff
@@ -0,0 +1,668 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables.impl;
+
+import com.google.common.base.Functions;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import io.pravega.client.tables.BadKeyVersionException;
+import io.pravega.client.tables.ConditionalTableUpdateException;
+import io.pravega.client.tables.MapWrapper;
+import io.pravega.client.tables.TableEntry;
+import io.pravega.client.tables.TableKey;
+import io.pravega.client.tables.Version;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.util.Retry;
+import java.util.AbstractCollection;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+
+/**
+ * Implementation of {@link MapWrapper}.
+ *
+ * @param <KeyT>   Key Type.
+ * @param <ValueT> Value Type.
+ */
+@RequiredArgsConstructor
+@SuppressWarnings({"unchecked", "NullableProblems"})
```

Comment:
This seems odd...

Reply:
Unchecked is because the Map API uses `Object` for arguments.

`NullableProblems` is because of annotations on the `Map` interface which the compiler complains about because it doesn't map those to the Lombok equivalents.